src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -166,7 +197,83 @@ internal void PerformBackup()
         internal BackupConfigInfo GetBackupConfigInfo(string databaseName)
         {
             return this.backupUtilities.GetBackupConfigInfo(databaseName);
-        }   
-        
+        }
+
+        /// <summary>
+        /// Create a backup task for execution and cancellation
+        /// </summary>
+        /// <param name="sqlTask"></param>
+        /// <returns></returns>
+        internal async Task<TaskResult> BackupTask(SqlTask sqlTask)
+        {
+            sqlTask.AddMessage(SR.Task_InProgress, SqlTaskStatus.InProgress, true);
+            Task<TaskResult> performTask = this.PerformTask();
+            Task<TaskResult> cancelTask = this.CancelTask(sqlTask);
+            Task<TaskResult> completedTask = await Task.WhenAny(performTask, cancelTask);
+            if (completedTask == performTask)
+            {
+                this.backupCompletedEvent.Reset();
```

Comment:
could you open an issue to track fixing this? currently the class seems like it can be easily used in a way that will cause it to not work correctly.

---
In reply to: [122376955](https://github.com/Microsoft/sqltoolsservice/pull/381#discussion_r122376955) [](ancestors = 122376955)